### PR TITLE
feat(core): add --affected to show only affected projects

### PR DESCRIPTION
**Pull request by AgentEnder**
This PR overhauls `nx show` a bit, to better support future object types and prepare for deprecating print-affected.

The main changes are:
- `nx show` is now a noop, but is built with subcommands
- `nx show projects` is a subcommand, and can take affected arguments
- `pnpm documentation` is equipped to handle subcommands

### CODE REVIEW

**Review by FrozenPandaz — APPROVED** [review]

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
Please fix e2e tests and add a new one for this.

File: docs/generated/cli/show.md [hunk 1]
```diff
@@ -10,11 +10,25 @@ Show information about the workspace (e.g., list of projects)
 ## Usage
 
 ```shell
-nx show <object>
+nx show
 ```
 
 Install `nx` globally to invoke the command directly using `nx`, or use `npx nx`, `yarn nx`, or `pnpm nx`.
 
+### Examples
```
Comment (FrozenPandaz) [inline docs/generated/cli/show.md:18]: Add an example for `nx show projects`

File: docs/generated/cli/show.md [hunk 2]
```diff
@@ -23,13 +37,73 @@ Type: `boolean`
...
+### version
 
-Choices: [projects]
+Type: `boolean`
 
-What to show (e.g., projects)
+Show version number
 
-### version
+## Child Commands
```
Comment (FrozenPandaz) [inline docs/generated/cli/show.md]:
```suggestion
## Sub Commands
```

File: packages/nx/src/command-line/examples.ts [hunk 3]
```diff
@@ -339,6 +339,18 @@ export const examples: Record<string, Example[]> = {
...
+      command: 'show projects --affected',
+      description: 'Show affected projects in the workspace',
+    },
+
+    {
+      command: 'show projects --affected --exclude *-e2e',
+      description:
+        'Show affected projects in the workspace, excluding end-to-end projects',
+    },
+  ],
```
Comment (FrozenPandaz) [inline packages/nx/src/command-line/examples.ts]:
```suggestion
  show: {
  projects: [
    {
      command: 'show projects --affected',
      description: 'Show affected projects in the workspace',
    },

    {
      command: 'show projects --affected --exclude *-e2e',
      description:
        'Show affected projects in the workspace, excluding end-to-end projects',
    },
  ]},
```

**Review by FrozenPandaz — APPROVED** [review]

File: packages/nx/src/command-line/examples.ts (hunk 3 above)
Comment (AgentEnder) [reply packages/nx/src/command-line/examples.ts]: We'll attack this in follow up :)

**Pull request merged**